supervisor/api/backups.py

```diff
@@ -484,6 +485,10 @@ async def upload(self, request: web.Request):
             if location and location != LOCATION_CLOUD_BACKUP:
                 tmp_path = location.local_where
 
+        filename: str | None = None
+        if ATTR_FILENAME in request.query:
```

Comment:
We need to validate this using the same `RE_BACKUP_FILENAME` regex we used here: 
https://github.com/home-assistant/supervisor/blob/61a2101d8ae77fbe75553cc4646d89423534bfc3/supervisor/api/backups.py#L104

Reply:
Right, added the check.